algo/src/main/java/org/neo4j/graphalgo/similarity/SimilarityExporter.java

```diff
@@ -74,8 +97,23 @@ private int getOrCreatePropertyId(String propertyName) {
                 .propertyKeyGetOrCreateForName(propertyName));
     }
 
-    private void writeSequential(Stream<SimilarityResult> similarityPairs) {
-        similarityPairs.forEach(this::export);
+    private void writeSequential(Stream<SimilarityResult> similarityPairs, long batchSize) {
```

Comment:
Would be really amazing if we could parallelize that.

Reply:
Yeh...I dunno if there's a way that we can partition the updates to run them in parallel without having any deadlocks.I suppose we need to create batches based on running connected components over the similarity graph that's going to be created?!

Reply:
Yep that's what I thought. Might be easier if we had the graph loaded to run the algo. Or we could load it temporarily to compute clusters then write the "inner" cluster rels in parallel first and then the "border" rels in a single batch.

Would be a fun experiment.

Reply:
We have this on another branch that's still WIP. I think we can merge this sequential batching and add the parallel writing afterwards